File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.swift
import Foundation

func solution(_ number:String, _ k:Int) -> String {
    let arr = number.map { String($0) } // ["1", "9", "2", "4"]
    var stack = [String]()
    var deleteCount = k
    
    for number in arr {
        while true {
            if stack.count == 0 {
                break
            } else {
                if stack[stack.count-1] < number && deleteCount > 0 {
                    deleteCount -= 1
                    stack.removeLast()
                } else {
                    break
                }
            }
        }
       
        stack.append(number)
    }
    
    // 만약 차마 못지운 것이 있다면 뒤에 있는 놈들부터 쭉 없앤다. 
    while deleteCount > 0 {
        stack.removeLast()
        deleteCount -= 1
    }

    return stack.joined()
}

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    answer = []
    for i in number:
        if len(answer) == 0:
            answer.append(i)
            continue
        if k > 0:
            while len(answer) > 0 and answer[-1] < i and k > 0:
                answer.pop()
                k -= 1
        answer.append(i)
    if k > 0: # 남은 게 있다면?..
        answer = answer[:-k]
        
    return "".join(answer)